File: week6/pset6/sentimental-readability/readability.py
from cs50 import get_string

text = get_string("Text: ")

letters = sentences = words = 0
i = 0
while True:
    if i >= len(text):
        break
    if text[i].isalpha():
        letters += 1
    elif text[i] in [".", "!", "?"]:
        words += 1
        sentences += 1
        i += 1
    elif text[i] in [',', ';']:
        words += 1
        i += 1
    elif text[i] == ' ':
        words += 1
    i += 1

L = letters * 100 / words
S = sentences * 100 / words

grade = round((0.0588 * L) - (0.296 * S) - 15.8)

if grade < 1:
    print("Before Grade 1")
elif grade >= 16:
    print("Grade 16+")
else:
    print(f"Grade {grade}")

File: week2/pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

void countLetterAndWord(string text);

int cLetter = 0, cWord = 0, cSentences = 0;

int main(void)
{
	string text = get_string("Text: ");
	//printf("%s\n", text);
    countLetterAndWord(text);
	//printf("cL:%i, cW:%i, cS:%i\n", cLetter, cWord, cSentences);

	//Coleman-Liau index
	long double L, S;
	L = ((long double)cLetter * 100) / (long double)cWord;
	S = ((long double)cSentences * 100) / (long double)cWord;
	int index = round((0.0588 * L) - (0.296 * S) - 15.8);
	if (index < 1)
	{
		printf("Before Grade 1\n");
	}
	else if (index >= 16)
	{
		printf("Grade 16+\n");
	}
	else
	{
	printf("Grade %i\n", index);
	}
	return 0;
}

void countLetterAndWord(string text)
{
	for (int count = 0, length = strlen(text); count < length; ++count)
	{
		if (isalpha(text[count]))
		{
			cLetter++;
		}
		else if (text[count] == '!' || text[count] == '?' || text[count] == '.' || text[count] == ',' || text[count] == ';')
		{
			cWord++;
			if (text[count] != ',' && text[count] != ';')
			{
				cSentences++;
			}
			count++;
		}
		else if (text[count] == ' ')
		{
			cWord++;
		}
	}
}
